elephant/utils.py

```diff
@@ -372,3 +394,54 @@ def get_opencl_capability():
         return True
     except ImportError:
         return False
+
+
+def trials_to_list_of_spiketrainlist(method):
+    """
+    Decorator to convert Trials object to a list of spiketrainlists before
+    calling the wrapped method.
+
+    Parameters
+    ----------
+    method: callable, The method to be decorated.
+
+    Returns
+    -------
+    callable: The decorated method.
+
+    Raises
+    ------
+    ValueError: If the number of trials in the Trials object is 0.
+
+    Example:
+        @trials_to_list_of_spiketrainlist
+        def process_data(self, spiketrains):
+            ...
+    """
+
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
```

Comment:
In my opinion, the decorator could be made generic by just having `*args`, `**kwargs` here. Then it would work for both pure functions or methods that have the object instance as the first argument, and be more reusable throughout Elephant. Any reason to have it just for methods?

Reply:
Thanks, good catch. This was a leftover the previous way of implementing this.
Fix: `self`- removed.